tests/BuildScriptGeneratorCli.Tests/BuildpackDetectCommandTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using System;
+using System.IO;
+using Xunit;
+using Microsoft.Oryx.Tests.Common;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.BuildScriptGeneratorCli;
+
+namespace BuildScriptGeneratorCli.Tests
+{
+    public class BuildpackDetectCommandTest : IClassFixture<TestTempDirTestFixture>
+    {
+        private readonly string _testDirPath;
+
+        public BuildpackDetectCommandTest(TestTempDirTestFixture testFixture)
+        {
+            _testDirPath = testFixture.RootDirPath;
+        }
+
+        [Fact]
+        public void Execute_OutputsNode_WhenPackageJsonExists()
+        {
+            // Arrange
+            var cmd = new BuildpackDetectCommand
+            {
+                SourceDir = _testDirPath,
+                PlatformDir = string.Empty,
+                PlanPath = string.Empty
+            };
+            var console = new TestConsole();
+            var svcProvider = GetServiceProvider(cmd);
+
+            // Act
+            int exitCode;
+
+            // Assert with an empty app directory
+            Assert.Equal(BuildpackDetectCommand.DetectorFailCode, cmd.Execute(svcProvider, console));
+
+            // Add file to app directory and assert again
+            File.WriteAllText(Path.Combine(_testDirPath, NodeConstants.PackageJsonFileName), "\n");
```

Comment:
Since `_testDirPath` is the root temporary directory for all the tests in this class, I would suggest to create a sub-dir under it to have this file. This way each test has its own sandboxed folder.